PDF.go not working.

Guys, still not working. (Issue related to #3 )

The error is importing the unidoc module.

In pdf.go, I have changed the imports:

~~~go
// pdf extracts metadata from pdf files
package pdf

import (
	"bytes"
	"github.com/datatogether/xmp"
	unicommon "github.com/unidoc/unidoc/common"
	unilicense "github.com/unidoc/unidoc/common/license"
	pdf "github.com/unidoc/unidoc/pdf/lib"
	"io"
~~~

instead of:

~~~go
// pdf extracts metadata from pdf files
package pdf

import (
	"bytes"
	"github.com/datatogether/xmp"
	unicommon "github.com/unidoc/unidoc/common" 
	unilicense "github.com/unidoc/unidoc/license"
	pdf "github.com/unidoc/unidoc/pdf"
	"io"
~~~

But the `unidoc/pdf` contains other folder inside.

When I try to run:

~~~bash
go run main.go
# github.com/datatogether/pdf
../../../go/src/github.com/datatogether/pdf/pdf.go:53:26: invalid operation: (*dict)["Type"] (type core.PdfObjectDictionary does not support indexing)
../../../go/src/github.com/datatogether/pdf/pdf.go:58:23: cannot range over *dict (type core.PdfObjectDictionary)
~~~

The main problem persists.

The problem is that the unidoc folder tree has changed inside the `pdf` directory I guess.

And also, from `pdf.go`:

~~~go
unilicense "github.com/unidoc/unidoc/license"
~~~

Needs to be changed to:

~~~go
unilicense "github.com/unidoc/unidoc/common/license"
~~~

In order to use the actual folder tree.

My go program:

~~~go
package main

import (
  "github.com/datatogether/pdf"
  //"github.com/datatogether/xmp"
  //"github.com/unidoc/unidoc/pdf/TEST"
  //"github.com/datatogether/xmp"
  //"testing"
  "fmt"
)

func main(){
  meta, err := MetadataForFile("test.pdf")
  fmt.Println(meta)
}
~~~

And when running:

~~~bash
go run .\pdf.go
# github.com/datatogether/pdf
..\go\src\github.com\datatogether\pdf\pdf.go:53:26: invalid operation: (*dict)["Type"] (type core.PdfObjectDictionary does not support indexing)
..\go\src\github.com\datatogether\pdf\pdf.go:58:23: cannot range over *dict (type core.PdfObjectDictionary)
~~~

--- a/pdf.go
+++ b/pdf.go
@@ -5,8 +5,8 @@ import (
 	"bytes"
 	"github.com/datatogether/xmp"
 	unicommon "github.com/unidoc/unidoc/common"
-	unilicense "github.com/unidoc/unidoc/license"
-	pdf "github.com/unidoc/unidoc/pdf"
+	unilicense "github.com/unidoc/unidoc/common/license"
+	pdf "github.com/unidoc/unidoc/pdf/core"
 	"io"
 	"os"
 )
@@ -43,22 +43,18 @@ func extract(r io.ReadSeeker) (map[string]interface{}, error) {
 		if err != nil || o.String() == "null" {
 			break
 		}
-
-		iobj, isIndirect := o.(*pdf.PdfIndirectObject)
-		if isIndirect {
+		if iobj, isIndirect := o.(*pdf.PdfIndirectObject); isIndirect {
 			// fmt.Printf("IND OOBJ %d: %s\n", xref.objectNumber, iobj)
-			dict, isDict := iobj.PdfObject.(*pdf.PdfObjectDictionary)
-			if isDict {
+			if dict, isDict := iobj.PdfObject.(*pdf.PdfObjectDictionary); isDict {
 				// Check if has Type parameter.
-				if ot, has := (*dict)["Type"].(*pdf.PdfObjectName); has {
-					otype := string(*ot)
+				if ot, has := dict.Get("Type").(*pdf.PdfObjectName); has {
 					// fmt.Printf("---> Obj type: %s\n", otype)
-					if otype == "Catalog" {
+					if ot.String() == "Catalog" {
 						// fmt.Println(dict.String())
-						for key, obj := range *dict {
+						for _, key := range dict.Keys() {
 							// TODO - check pdf spec, is only one metadata entry allowed?
 							if key.String() == "Metadata" {
-								oNum := obj.(*pdf.PdfObjectReference).ObjectNumber
+								oNum := dict.Get(key).(*pdf.PdfObjectReference).ObjectNumber
 								obj, err := p.LookupByNumber(int(oNum))
 								if err != nil {
 									return nil, err

